src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1927 @@
+                        return new RegexSimpleEscapeNode(
+                            backslashToken, ConsumeCurrentToken(allowTrivia: false).With(kind: RegexKind.TextToken));
+
+                    default:
+                        // trivia is not allowed anywhere in a character class
+                        _lexer.Position--;
+                        return ScanCharEscape(backslashToken, allowTriviaAfterEnd: false);
+                }
+            }
+
+            if (!afterRangeMinus &&
+                !isFirst &&
+                _currentToken.Kind == RegexKind.MinusToken &&
+                _lexer.IsAt("["))
+            {
+                // have a trailing subtraction.
+                // trivia is not allowed anywhere in a character class
+                return ParseCharacterClassSubtractionNode(
+                    ConsumeCurrentToken(allowTrivia: false));
+            }
+
+            // From the .net regex code:
+            // This is code for Posix style properties - [:Ll:] or [:IsTibetan:].
+            // It currently doesn't do anything other than skip the whole thing!
+            if (!afterRangeMinus && _currentToken.Kind == RegexKind.OpenBracketToken && _lexer.IsAt(":"))
+            {
+                var beforeBracketPos = _lexer.Position - 1;
+                // trivia is not allowed anywhere in a character class
+                ConsumeCurrentToken(allowTrivia: false);
+
+                var captureName = _lexer.TryScanCaptureName();
+                if (captureName.HasValue && _lexer.IsAt(":]"))
+                {
+                    _lexer.Position += 2;
+                    var textChars = _lexer.GetSubPattern(beforeBracketPos, _lexer.Position);
+                    var token = CreateToken(RegexKind.TextToken, ImmutableArray<RegexTrivia>.Empty, textChars);
+
+                    // trivia is not allowed anywhere in a character class
+                    ConsumeCurrentToken(allowTrivia: false);
+                    return new RegexPosixPropertyNode(token);
+                }
+                else
+                {
+                    // Reset to back where we were.
+                    // trivia is not allowed anywhere in a character class
+                    _lexer.Position = beforeBracketPos;
+                    ConsumeCurrentToken(allowTrivia: false);
+                    Debug.Assert(_currentToken.Kind == RegexKind.OpenBracketToken);
+                }
+            }
+
+            // trivia is not allowed anywhere in a character class
+            return new RegexTextNode(
+                ConsumeCurrentToken(allowTrivia: false).With(kind: RegexKind.TextToken));
+        }
+
+        private RegexPrimaryExpressionNode ParseCharacterClassSubtractionNode(RegexToken minusToken)
+        {
+            var charClass = ParseCharacterClass();
+
+            if (_currentToken.Kind != RegexKind.CloseBracketToken && _currentToken.Kind != RegexKind.EndOfFile)
+            {
+                minu
```

Comment:
Same comment here. Probably good to make the same change you made above (add comment and use 'A' instead).

Reply:
sorry, not sure what this is referring to.